src/field.rs

```diff
@@ -531,29 +531,20 @@ pub fn merge_vector<F: FieldElement>(
     Ok(())
 }
 
-/// Generate a vector of uniform random field elements.
-pub fn rand<F: FieldElement>(len: usize) -> Result<Vec<F>, getrandom::Error> {
-    Ok(Prng::new_with_length(len)?.collect())
-}
-
 /// Outputs an additive secret sharing of the input.
-pub fn split<F: FieldElement>(
-    inp: &[F],
-    num_shares: usize,
-) -> Result<Vec<Vec<F>>, getrandom::Error> {
+pub fn split<F: FieldElement>(inp: &[F], num_shares: usize) -> Result<Vec<Vec<F>>, PrngError> {
     if num_shares == 0 {
         return Ok(vec![]);
     }
 
-    let mut outp = vec![vec![F::zero(); inp.len()]; num_shares];
-    outp[0][..inp.len()].clone_from_slice(inp);
+    let mut outp = vec![vec![]; num_shares];
+    outp[0] = inp.to_vec();
 
-    let mut prng = Prng::new()?;
     for i in 1..num_shares {
-        for j in 0..inp.len() {
-            let r = prng.next().unwrap();
-            outp[i][j] = r;
-            outp[0][j] -= r;
+        let r: Vec<F> = random_vector(inp.len())?;
+        outp[i] = r.clone();
+        for (x, y) in outp[0].iter_mut().zip(r) {
+            *x -= y;
         }
```

Comment:
```
error[E0502]: cannot borrow `outp` as immutable because it is also borrowed as mutable
   --> src/field.rs:545:47
    |
545 |         for (x, y) in outp[0].iter_mut().zip(&outp[i]) {
    |                       ----               ---  ^^^^ immutable borrow occurs here
    |                       |                  |
    |                       |                  mutable borrow later used by call
    |                       mutable borrow occurs here

```

Reply:
Found a different way, but I'm not sure it avoids a copy.

Reply:
What you've got now avoids the `clone()` so the unnecessary copy is gone.